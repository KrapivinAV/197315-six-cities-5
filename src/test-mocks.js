export const offerFirstCityIsFavoriteIsPremiumHostIsPro = {
  city: {
    name: `Paris`,
    location: {
      latitude: 48.85661,
      longitude: 2.351499,
      zoom: 13
    }
  },
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  title: `The house among olive`,
  rating: 3,
  type: `apartment`,
  bedrooms: 5,
  price: 226,
  goods: [
    `Dishwasher`,
    `Fridge`,
    `Towels`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    avatarUrl: `/img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything.`,
  location: {
    latitude: 48.87761,
    longitude: 2.333499,
    zoom: 16
  },
  id: 1,
  isFavorite: true,
  isPremium: true,
  maxAdults: 8
};

export const offerFirstCityNotFavoriteNotPremiumHostNotPro = {
  city: {
    name: `Paris`,
    location: {
      latitude: 48.85661,
      longitude: 2.351499,
      zoom: 13
    }
  },
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  title: `The house among olive`,
  rating: 3,
  type: `apartment`,
  bedrooms: 5,
  price: 226,
  goods: [
    `Dishwasher`,
    `Fridge`,
    `Towels`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    avatarUrl: `/img/avatar-angelina.jpg`,
    isPro: false
  },
  description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything.`,
  location: {
    latitude: 48.87761,
    longitude: 2.333499,
    zoom: 16
  },
  id: 2,
  isFavorite: false,
  isPremium: false,
  maxAdults: 8
};

export const offerSecondCityIsFavoriteNotPremiumHostNotPro = {
  city: {
    name: `Paris`,
    location: {
      latitude: 48.85661,
      longitude: 2.351499,
      zoom: 13
    }
  },
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  title: `The house among olive`,
  rating: 3,
  type: `apartment`,
  bedrooms: 5,
  price: 226,
  goods: [
    `Dishwasher`,
    `Fridge`,
    `Towels`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    avatarUrl: `/img/avatar-angelina.jpg`,
    isPro: false
  },
  description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything.`,
  location: {
    latitude: 48.87761,
    longitude: 2.333499,
    zoom: 16
  },
  id: 3,
  isFavorite: true,
  isPremium: false,
  maxAdults: 8
};

export const REVIEW_FIRST = {
  id: 1,
  rating: 3,
  comment: `99999999999999999999999999999999999999999999999999999`,
  date: `2020-11-30T06:28:10.586Z`,
  user: {
    id: 1,
    name: `ak`,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg`,
    isPro: true
  }
};

export const REVIEW_SECOND = {
  id: 2,
  rating: 5,
  comment: `99999999999999999999999999999999999999999999999999999`,
  date: `2020-10-30T06:28:10.586Z`,
  user: {
    id: 1,
    name: `ak`,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg`,
    isPro: false
  }
};
