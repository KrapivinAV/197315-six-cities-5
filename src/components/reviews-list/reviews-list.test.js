import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list";
import {REVIEW_FIRST, REVIEW_SECOND} from "../../test-mocks";

describe(`Render ReviewsList`, () => {
  it(`Render ReviewsList`, () => {

    const tree = renderer
      .create(
          <ReviewsList
            offerReviews={[REVIEW_FIRST, REVIEW_SECOND]}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
