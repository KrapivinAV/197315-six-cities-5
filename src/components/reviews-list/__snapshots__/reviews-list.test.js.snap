// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ReviewsList Render ReviewsList 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        ak
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        99999999999999999999999999999999999999999999999999999
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        November 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        ak
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        99999999999999999999999999999999999999999999999999999
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        October 2020
      </time>
    </div>
  </li>
</ul>
`;
